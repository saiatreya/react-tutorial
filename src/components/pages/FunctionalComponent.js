import React from 'react';
import Button from '../styled/Button';

const FunctionalComponent = (props) => {
    return (
        <div>
            <h1>Functional Components: </h1>
            <ul>
                <li>Functional components are used to be stateless components. But with the introduction of hooks functional components also have state.</li>
                <li>Functional components takes props parameter from another class/functional component.</li>
            </ul>
            <h2># Taking value from parent as props: </h2>
            <p>From parent <i>ClassComponent.js</i> we are passing name to functional component</p>
            <i>Name: </i>{props.name || 'No Name passed from parent'}
            <h2># Counter value from Parent: </h2>
            <i>Counter from parent: {props.counter}</i>
            <p>Click on the above <i>"Click to Increment Counter"</i> button to check Counter value incremented based on parent counter</p>
            <h2># Counter value incrementing from child: </h2>
            <p>Click on the below button to increment the counter in both child and parent at the same time. Here counter incrementing functionality has been passed as a props to child.</p>
            <Button onClick={props.counterCallBack}>Click to increment counter in both child and parent.</Button>
            <br /><br />
            <div className="file-name">Filename is >>>>> <strong>FunctionalComponent.js</strong></div>
        </div>
    );
};

export default FunctionalComponent;