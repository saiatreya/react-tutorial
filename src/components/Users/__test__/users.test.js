import React from 'react';
import ReactDOM from 'react-dom';
import FunctionalComponent from '../../pages/FunctionalComponent';
import renderer from 'react-test-renderer';
describe("Simple test suite", () => {
    it("Test 1", () => {
        const div = document.createElement('DIV');
        ReactDOM.render(
            <FunctionalComponent />
            , div);
    });
    it("matches snapshot", () => {
        const tree = renderer.create(<FunctionalComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});