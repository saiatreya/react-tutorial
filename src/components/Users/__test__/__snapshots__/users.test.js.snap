// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple test suite matches snapshot 1`] = `
<div>
  <h1>
    Functional Components: 
  </h1>
  <ul>
    <li>
      Functional components are used to be stateless components. But with the introduction of hooks functional components also have state.
    </li>
    <li>
      Functional components takes props parameter from another class/functional component.
    </li>
  </ul>
  <h2>
    # Taking value from parent as props: 
  </h2>
  <p>
    From parent 
    <i>
      ClassComponent.js
    </i>
     we are passing name to functional component
  </p>
  <i>
    Name: 
  </i>
  No Name passed from parent
  <h2>
    # Counter value from Parent: 
  </h2>
  <i>
    Counter from parent: 
  </i>
  <p>
    Click on the above 
    <i>
      "Click to Increment Counter"
    </i>
     button to check Counter value incremented based on parent counter
  </p>
  <h2>
    # Counter value incrementing from child: 
  </h2>
  <p>
    Click on the below button to increment the counter in both child and parent at the same time. Here counter incrementing functionality has been passed as a props to child.
  </p>
  <button
    className="sc-bdfBwQ dsaiBp"
  >
    Click to increment counter in both child and parent.
  </button>
  <br />
  <br />
  <div
    className="file-name"
  >
    Filename is &gt;&gt;&gt;&gt;&gt; 
    <strong>
      FunctionalComponent.js
    </strong>
  </div>
</div>
`;
